### PR TITLE
Plugin Geomap: Migration to OpenLayers v3.x

### PR DESCRIPTION
**Pull request by jvanulde**

### CODE REVIEW

File: bower.json
Comment (nschonni) [inline bower.json]: Do we still need a fork for this, or can we just use the regular NPM/Bower package now?

File: src/i18n/base.js
Comment (nschonni) [inline src/i18n/base.js]: why is this changing, and deprecated how?

File: src/plugins/deps/geomap-lib.js
Comment (nschonni) [inline src/plugins/deps/geomap-lib.js]: Without renaming the parameter, the diff would be shrunk

File: src/plugins/deps/geomap-lib.js
Comment (nschonni) [inline src/plugins/deps/geomap-lib.js]: if the spaces are to avoid the substring, you could just convert className to an array first and then use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf

File: src/plugins/geomap/geomap.js
Comment (nschonni) [inline src/plugins/geomap/geomap.js]: What are the inline styled `div`s for?

File: bower.json
Comment (jvanulde) [inline bower.json]: I don't know. I would have to test the npm package.

File: src/i18n/base.js
Comment (jvanulde) [inline src/i18n/base.js]: It's for backwards compatibility. We can eventually remove it.

File: src/plugins/geomap/geomap.js
Comment (jvanulde) [inline src/plugins/geomap/geomap.js]: Purely for UX. They are interface skeletons. Without it we would probably want a spinner but since the map element already as one it would be confusing to have two spinners. So we have a skeleton in the legend element while the map is loading.

Comment (duboisp) [inline src/plugins/geomap/geomap.js]: Do you think it is possible to move the inline style into the [geomap SASS file](https://github.com/wet-boew/wet-boew/blob/master/src/plugins/geomap/_base.scss)?

Comment (jvanulde) [inline src/plugins/geomap/geomap.js]: Not sure. Is it something that needs to be resolved? I don't understand why it's a problem.

File: bower.json
Comment (LaurentGoderre) [inline bower.json]: @jvanulde can we stick with the old repo and move the tags and the branch to the openlayers-dist repo?

File: src/plugins/geomap/geomap.js
Comment (duboisp) [inline src/plugins/geomap/geomap.js]: @jvanulde as discuss at the PR review meeting, it's a best practice to **not use inline style**, so we kindly asking you to use CSS class name instead and then put the custom CSS into the [geomap SASS file](https://github.com/wet-boew/wet-boew/blob/master/src/plugins/geomap/_base.scss)

File: src/i18n/base.js
Comment (LaurentGoderre) [inline src/i18n/base.js]: @jvanulde this shouldn't affect compatibility because the i18n file is tied to a WET version. You shouldn't use the v4.0.12 i18n file with v4.0.24 for example.

File: src/plugins/geomap/geomap.js
Comment (lucas-hay) [inline src/plugins/geomap/geomap.js]: @jvanulde Are you able to do Pierre's request above? We will merge once that is completed.

File: bower.json
Comment (jvanulde) [inline bower.json]: So what do we need to here? I want to move forward with this PR. Let me know what the best course of action is.

Comment (jvanulde) [inline bower.json]: @nschonni I can reference https://github.com/openlayers/openlayers.git#3.17.1 but how do I npm build it after npm install? I need the distribution lib.

Comment (jvanulde) [inline bower.json]: I was able to do this by adding it as a devDependency. Not sure that was the best way to do it.

File: src/i18n/base.js
Comment (jvanulde) [inline src/i18n/base.js]: Okay.

File: src/plugins/deps/geomap-lib.js
Comment (jvanulde) [inline src/plugins/deps/geomap-lib.js]: Okay. I will rename it.

File: src/plugins/deps/geomap-lib.js
Comment (jvanulde) [inline src/plugins/deps/geomap-lib.js]: Good idea.

**Comment by lucas-hay** [comment]
@jvanulde We are having a PR review meeting tomorrow and would like to get this merged for you.
Can you please fix the merge conflict. Also, could you push this to a gh-pages branch so we have a working example to test?
If you would like to joing to PR review meeting @10:00 am tomorrow morning, reply here and I will forward the meeting request to you.

**Comment by EricDunsworth** [comment]
@jvanulde Would it be possible to remove the merge commit from this PR? It looks like it was added yesterday.

**Comment by jvanulde** [comment]
@EricDunsworth what's the best way to remove the merge commit? Do a hard reset?

**Comment by duboisp** [comment]
@jvanulde 
You can do a hard reset to your last commit 9e4ee27e96b06ca350bc49c416c91444a66f36ea
Fetch the upstream (wet-boew/wet-boew.git)
Then you do a rebase upstream/wet-boew

That should work

**Comment by shawnthompson** [comment]
I created a working example of the PR.

https://servicecanada.github.io/wet-boew-jvanulde-geomap-improvements/demos/geomap/geomap-en.html

Can we test that the change works.

**Comment by jvanulde** [comment]
That's the old version @shawnthompson.

**Comment by shawnthompson** [comment]
@jvanulde just checked out this PR like github told me to do and ran `grunt dist` and put the dist folder in a gh-pages branch to a new repo.

**Comment by shawnthompson** [comment]
Updated
https://servicecanada.github.io/wet-boew-jvanulde-geomap-improvements/demos/geomap/geomap-en.html

**Comment by duboisp** [comment]
Is that service outage at NRCan cause the "mixed (insecure)" error in the FF console and showing a Security alert in IE?

geomap-2017-07-27

**Comment by lucas-hay** [comment]
It also happens in FF.
Joost can you make the call to that file "protocolless"
//geoappext.nrcan.gc.ca/arcgis/rest/services/BaseMaps/provinces1c/MapServer/export?F=image&FORMAT=PNG32&TRANSPARENT=true&SIZE=256%2C256&BBOX=0%2C0%2C5003769.342810653%2C5003769.342810653&BBOXSR=3978&IMAGESR=3978&DPI=90

instead of:

http://geoappext.nrcan.gc.ca/arcgis/rest/services/BaseMaps/provinces1c/MapServer/export?F=image&FORMAT=PNG32&TRANSPARENT=true&SIZE=256%2C256&BBOX=0%2C0%2C5003769.342810653%2C5003769.342810653&BBOXSR=3978&IMAGESR=3978&DPI=90

**Comment by jvanulde** [comment]
@lucas-hay it actually is in the i18n file: geo-bmap-url,//geoappext.nrcan.gc.ca/arcgis/rest/services/BaseMaps/CBMT3978/MapServer/WMTS/

**Comment by lucas-hay** [comment]
@jvanulde We plan to do a release on August 15th, is this PR ready?
Can you address the mixed content issue by updating the reference as mentioned above?
We would like to see this PR in the release next week

**Comment by jvanulde** [comment]
Well, I am loading "//geoappext.nrcan.gc.ca/arcgis/rest/services/BaseMaps/CBMT3978/MapServer/WMTS/" so the URI is protocolless. I am not sure why we are getting a mixed content issue...

File: src/i18n/base.js
Comment (jvanulde) [reply src/i18n/base.js]: @LaurentGoderre it looks it did affect compatibility because all the existing maps are now broken: http://wet-boew.github.io/wet-boew/demos/geomap/geomap-en.html

Either we need to add that key back or merge this PR. Thoughts?

**Comment by jvanulde** [comment]
@EricDunsworth, @duboisp, @shawnthompson, @lucas-hay any chance that this will be merged for next release? There are breaking changes due to removal of backwards compatibility in i18n.csv.

**Comment by EricDunsworth** [comment]
@jvanulde I'm just a contributor (not a maintainer), so I lack the capability of merging in PRs.

Btw it still looks like this PR contains a merge commit. I'd recommend removing it and rebasing your real commits on top of upstream/master.

As far as the mixed content issue goes, has that been fully resolved?

From what I can tell from this PR's discussion, those are the only two remaining issues that have held back this PR back from getting merged in.

**Comment by jvanulde** [comment]
@EricDunsworth the mixed content has been resolved. Not sure what the merge commit is all about or how to resolve it.

**Comment by EricDunsworth** [comment]
@jvanulde Thanks for clarifying that the mixed content issue was resolved :)!

As far as the merge commit goes, the commit I'm referring to is 3931b2ae98ef33ac6d88b129198c296d1ecafa6b. Its unrelated to the rest of the commits in this pull request and the line it added already exists in wet-boew/master.

To remove the merge commit, I'd recommend retrying the steps @duboisp mentioned in https://github.com/wet-boew/wet-boew/pull/7889#issuecomment-313428278 against your latest "real" commit (59dee41d1dc3850609426b6cd009d00753be86af).

In Git, if you add commits to a branch, then pull the latest commits from the upstream/master remote, a merge commit will be automatically created if upstream/master contains commits that don't already exist in your branch. This might not be the best way of explaining it, but it basically "combines" your branch and upstream/master's commit timelines together.

In the context of a project like WET, IMO merge commits make sense if it's the result of merging a pull request into wet-boew/master. But it isn't ideal for a pull request like this one to bring unrelated merge commits (like "Merge branch 'master' into [branch]") into the project.

To avoid undesirable merge commits, I'd recommend against pulling from upstream/master while within a feature branch that contains WIP commits. If you need to bring upstream/master's latest commits into your feature branch without inadvertently introducing a merge commit, https://github.com/wet-boew/wet-boew/pull/7889#issuecomment-313428278 is likely to be your best bet.

Btw you don't need to always keep your feature branches up-to-date with upstream/master. It's usually only worthwhile if a PR based on your feature branch has conflicts according to GitHub or if you suspect that your changes might be incompatible or need to be revised as a result of really recent changes made to upstream/master.

Related resources:
* https://git-scm.com/docs/git-merge
* https://git-scm.com/docs/git-rebase
* https://git-scm.com/book/en/v2/Git-Branching-Rebasing

**Comment by jvanulde** [comment]
@EricDunsworth shouldn't I be rebasing on master after resetting my "real" commit?

**Comment by EricDunsworth** [comment]
@jvanulde Earlier today I added your wet-boew fork as a remote, fetched your geomap-improvements branch and experimented.

I believe the git commands below should be able to successfully clean up this PR's commits to only consist of the 4 "real" commits you made. 

Before starting, make sure you don't have any WIP changes or commits in your master branch that you don't want to lose, since the commands below will erase them.

```bash
# Check which version of git you're using.
git --version

# Don't proceed unless you're on git 1.7.3 or higher.

# Make your local master branch identical to the official wet-boew repo's master branch.
git checkout master
git fetch upstream
git reset --hard upstream/master

# Just in case anything goes horribly wrong...
# Create a backup branch of geomap-improvements.
git checkout -b backup-geomap-improvements geomap-improvements

# Switch over to your geomap-improvements branch.
git checkout geomap-improvements

# Give preference to the master branch when a merge conflict is encountered in 
# Gruntfile.coffee.
# It'll cause the rebase operation to automatically resolve conflicts in favour of the 
# master branch without any intervention on your part.
git rebase --strategy recursive -X ours master

# Force push the rebased branch to origin.
# It'll rewrite this PR's commits to only leave your "real" commits in place, but can 
# potentially be dangerous if the rebase operation messed up your changes.
git push origin geomap-improvements --force
```

**Comment by lucas-hay** [comment]
@EricDunsworth @duboisp @nschonni @LaurentGoderre 
Is this good to go now guys?

**Comment by duboisp** [comment]
@jvanulde is there a reason I can't run it locally, the map didn't show and I have the following error in the console:

geomap

----
Side note: This PR introduce some change to the i18n, after this PR are approved and merged, we would need to update the i18n spreedsheet.

**Comment by jvanulde** [comment]
You will need the updated i18n for sure. I just checked and it runs locally for me...

**Comment by duboisp** [comment]
See working example from my build:

http://universallabs.org/wet/labs/geomap-nov-3-2017/unmin/demos/geomap/geomap-en.html

**Comment by jvanulde** [comment]
@duboisp that's the old version. Should look like: https://servicecanada.github.io/wet-boew-jvanulde-geomap-improvements/demos/geomap/geomap-en.html

**Comment by duboisp** [comment]
@jvanulde I hen created a working example from the commits in this PR. Is this PR contains the latest change?

**Comment by duboisp** [comment]
Wait a minutes, I didn't re-run the wet-boew setup. I doing that right now.

**Comment by duboisp** [comment]
I cleared my npm and bower cache and fully rebuild, but I still get the same error message. @jvanulde can you look if this PR, your branch, include all the latest changes? Thanks

**Comment by jvanulde** [comment]
I am looking at the diffs and they include the latest changes. If I look at your working example it still contains all the old code...

**Comment by duboisp** [comment]
Ok, I fully resetting my local repro, did a rebase and now it works

http://universallabs.org/wet/labs/geomap-nov-3-2017/unmin/demos/geomap/geomap-en.html

Thanks @jvanulde

File: src/plugins/geomap/_base.scss
Comment (EricDunsworth) [inline src/plugins/geomap/_base.scss]: @jvanulde I presume this change was made to avoid https://travis-ci.org/wet-boew/wet-boew/builds/296480431#L2266?

I'd recommend removing that line rather than commenting it out. There isn't any value in keeping it in a commented-out state in the long run.

Btw I believe WET currently only officially supports vendor prefixes for Webkit/Blink (``-webkit-``), Firefox (``-moz``) and IE/Edge (``-ms-``). AFAIK Konqueror (``-khtml-``) and classic Opera (``-o-``) prefixes aren't supported. Maybe @LaurentGoderre or @nschonni could definitively confirm.

**Review by EricDunsworth — CHANGES_REQUESTED** [review]
Remove ``-khtml-user-select: none;`` instead of commenting it out.

**Comment by nschonni** [comment]
Yes, vendor prefixing is handled by Autoprefixer already during the build based off a list of supported browsers                                                                                                                                                                                                                                                                                                                                        Sent from my BlackBerry 10 smartphone.                                                                                                                                                                                                                From: EricDunsworthSent: Wednesday, November 8, 2017 4:15 PMTo: wet-boew/wet-boewReply To: wet-boew/wet-boewCc: Nick Schonning; MentionSubject: Re: [wet-boew/wet-boew] Plugin Geomap: Migration to OpenLayers v3.x (#7889)@EricDunsworth requested changes on this pull request.

Remove -khtml-user-select: none; instead of commenting it out.

In src/plugins/geomap/_base.scss:
> @@ -279,7 +279,7 @@
 }
 
 .ol-viewport .ol-unselectable {
-	-khtml-user-select: none;
+	/* -khtml-user-select: none; sasslint doesn't like this */

@jvanulde I presume this change was made to avoid https://travis-ci.org/wet-boew/wet-boew/builds/296480431#L2266?
I'd recommend removing that line rather than commenting it out. There isn't any value in keeping it in a commented-out state in the long run.
Btw I believe WET currently only officially supports vendor prefixes for Webkit/Blink (-webkit-), Firefox (-moz) and IE/Edge (-ms-). AFAIK Konqueror (-khtml-) and classic Opera (-o-) prefixes aren't supported. Maybe @LaurentGoderre or @nschonni could definitively confirm.

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/wet-boew/wet-boew","title":"wet-boew/wet-boew","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/wet-boew/wet-boew"}},"updates":{"snippets":[{"icon":"PERSON","message":"@EricDunsworth requested changes on #7889"}],"action":{"name":"View Pull Request","url":"https://github.com/wet-boew/wet-boew/pull/7889#pullrequestreview-75250246"}}}

**Comment by EricDunsworth** [comment]
@lucas-hay FYI all the things I requested changes for have been addressed.

The "@EricDunsworth requested changes" message in the "Reviewers" panel can be ignored. I wanted to dismiss my review to make that stop appearing... But GitHub's interface doesn't currently allow non-maintainers to dismiss their own reviews (since they don't count towards mergeability). I didn't want to override it with an "Approved" review since I haven't gone through the whole PR in detail.

**Comment by lucas-hay** [comment]
@duboisp, @LaurentGoderre, @nschonni are you guys good with this now?

**Comment by duboisp** [comment]
I created a new working example with the latest change of this PR

Demo with all the updates: http://universallabs.org/wet/labs/geomap-nov-21-2017/unmin/demos/geomap/geomap-en.html

WET 4.0.26 working example but that use the latest geomap code: http://universallabs.org/wet/labs/geomap-nov-21-2017/unmin/demos/geomap-wet-4.0.26/geomap-en.html

@jvanulde an existing map that was working in WET 4.0.26 seems to not work with all the code change you made. So for me that may means every page that currently use geomap in a prior version of wet-boew will require an manual update. Can you describe what need to be updated? Is there a way to continue to support map that was configured like the working example in WET 4.0.26?

**Review by duboisp — CHANGES_REQUESTED** [review]
Ensure that geomap working example configured for WET 4.0.26 and prior version remain compatible and remain to be a valid configuration.

**Comment by jvanulde** [comment]
@duboisp we had to fix a non-standard supporting of WKT points that had a comma between the coordinate values in the sample dataset. We can make it backward compatible but ideally, implementers will change their WKT coordinate pairs in HTML tables to conform to the standard.

**Comment by duboisp** [comment]
Do you mind to make it backward compatible and continue to support the standard WKT points? Doing that will provide a transition period when implementer will update to WET 4.0.27.  Also include a note in the code to remove that "backward compatibility fix" at the next major version of wet.

@lucas-hay it's your call, but it might worth it to include a note in the release note about it.

**Comment by duboisp** [comment]
Thanks for the update @jvanulde 

I updated my working example with your latest code
* New working example: http://universallabs.org/wet/labs/geomap-nov-21-2017/unmin/demos/geomap/geomap-en.html
* WET 4.0.26 working example: http://universallabs.org/wet/labs/geomap-nov-21-2017/unmin/demos/geomap-wet-4.0.26/geomap-en.html

Now all the major map render as expected but I still have two map that don't work.
* Static map example 
* OSM basemap example

Is that something it could be made backward compatible?

**Comment by jvanulde** [comment]
@duboisp Excellent. Unfortunately, the other maps can't be made backward compatible due to non-existent services (OSM base map) and changes in the OL library (Static map). We provide updated examples.

**Comment by duboisp** [comment]
@jvanulde is that documented? Can a note be directed added near those working example.

**Comment by nschonni** [comment]
Nothing about the code or what we've reviewed, but I don't think this should land in a bug-fix release.
Something like this should ideally land in a 4.1 or 5.0 depending on the just features vs breaking changes.

**Comment by jvanulde** [comment]
@duboisp @nschonni if we don't merge this PR now then I will need to add a backward compatibility fix that was removed. Right now every Geomap implementation that uses the default configuration has not base map.

**Comment by duboisp** [comment]
@jvanulde how long it will take you to do that backward compatibility fix? Is that something you can complete today?

**Comment by jvanulde** [comment]
@duboisp its just a change to the i18n csv. I can do that now.

**Comment by duboisp** [comment]
What is that change? I would apply it to the i18n spreadsheet.

**Comment by jvanulde** [comment]
I added it to Google Docs but here it is...

geo-bmapurltxt:
en:
//geoappext.nrcan.gc.ca/arcgis/rest/services/BaseMaps/CBMT_TXT_3978/MapServer/WMTS/tile/1.0.0/BaseMaps_CBMT3978/{Style}/{TileMatrixSet}/{TileMatrix}/{TileRow}/{TileCol}.jpg 
fr:
//geoappext.nrcan.gc.ca/arcgis/rest/services/BaseMaps/CBCT_TXT_3978/MapServer/WMTS/tile/1.0.0/BaseMaps_CBMT3978/{Style}/{TileMatrixSet}/{TileMatrix}/{TileRow}/{TileCol}.jpg 

geo-bmapurl: //geoappext.nrcan.gc.ca/arcgis/rest/services/BaseMaps/CBMT_CBCT_GEOM_3978/MapServer/WMTS/tile/1.0.0/BaseMaps_CBMT3978/{Style}/{TileMatrixSet}/{TileMatrix}/{TileRow}/{TileCol}.jpg

**Comment by duboisp** [comment]
@jvanulde Great so I will get the google docs in sync with wet-boew

**Review by duboisp — APPROVED** [review]

**Comment by duboisp** [comment]
@jvanulde I changed the i18n english file in my working example and the I got the following result

* OSM Basemap example do not work
* The Static Map example is always loading, but after a while a canada map is displayed

Also there is a few error in the console, some I think is because of CORS request and one saying map.getlayer is not a function.

See: http://universallabs.org/wet/labs/geomap-nov-21-2017/unmin/demos/geomap-wet-4.0.26/geomap-en.html

**Comment by jvanulde** [comment]
@duboisp Your http://universallabs.org/wet/labs/geomap-nov-21-2017/unmin/demos/geomap-wet-4.0.26/demo/config-en.js is out of date.

**Comment by duboisp** [comment]
> Your http://universallabs.org/wet/labs/geomap-nov-21-2017/unmin/demos/geomap-wet-4.0.26/demo/config-en.js is out of date

I re-used exactly the one that was provided in the WET 4.0.26 working example. I assume that implementer of geomap may have follow what how it was demo'ed.

So may be it should have some explicit instruction on how to migrate those old config to the new config. But I will hope that old config will still work after this PR get merged.

**Comment by lucas-hay** [comment]
@jvanulde @duboisp Any news on this for tomorrow's release?

**Comment by duboisp** [comment]
@lucas-hay it will be to know if @jvanulde can make those broken think "backward compatible". At least from a geopmap plugin configuration perspective.

Or @lucas-hay it's up to you if you want to introduce a breaking change. But if you go in that direction, I would recommend at least: To ensure all the "breaking change" is documented in the geomap documentation. To ensure there is an explicit note box (alert-warning) beside each the working example that will became broken along with instruction on how to get it fix. And to add an explicit note directly in the release note of WET about this.

**Comment by jvanulde** [comment]
@duboisp @lucas-hay let us know what issues still remain. We can't solve the not supported formats and non-existent services issues. Those are out of anyone's control. Besides those were just demo's showing how you could bring in those service/file-types. Since they are no longer supported or available we have replaced them alternatives - again just to show how a variety of service/file-types can be used. Anything else we can/should deal with before release, but I don't see anything "breaking" right now unless I have missed something.

**Comment by duboisp** [comment]
@jvanulde Are you able to document that in the working example and in the documentation. Also, in the documentation, include a section that explain what has been change and what need to be done in order to  migrate it. Like explaining how you migrated the working example that didn't worked to a working example that now work.

**Comment by lucas-hay** [comment]
@jvanulde If you can complete what Pierre is suggesting before tomorrow, it can be included in the release. The release has pushed until tomorrow.

**Comment by jvanulde** [comment]
@duboisp I will update the documentation but I don't have anyone available for translation. As well, I don't think it makes sense to update the working examples to include a description of how the old previous example(s) no longer work(s). The old example doesn't work because the service doesn't exist anymore. The replacement example is a new service type altogether. Essentially, it can't be fixed because it's gone.

**Comment by lucas-hay** [comment]
@jvanlude, there is also a conflict so you will have to rebase.

**Comment by duboisp** [comment]
@jvanulde in the working example you can simply include a alert warning box that said, like you said "Since WET 4.0.27 we updated geomap to use to Open Layer 3 and the services that was used to create the following working example has been changed. Please consult the geomap documentation (with a direct link) to know how you can migrate your pages that was created based on the following working example."
Then in the geomap documentation, you explain how a web authors can migrate (step by step) their existing pages to use the new services.

Regarding the translation, the documentation can stay in English, it will only require you to replicated in the French docs. Regarding translating the information boxes into French, I can do it.

File: bower.json
Comment (EricDunsworth) [inline bower.json]: There's currently a merge conflict on this line. PR #8101 added a line directly before it (see https://github.com/wet-boew/wet-boew/pull/8101/files#diff-0a08a7565aba4405282251491979bb6b).

To fix the conflict, this PR will need to be rebased again and the conflict will need to be manually resolved. I could be mistaken, but I don't believe any extra options can be used with ``git rebase`` to automatically resolve the conflict while retaining both changes.

**Comment by duboisp** [comment]
Working example with the merge conflict resolved and with the latest commits

http://universallabs.org/wet/labs/geomap-dec-1-2017/unmin/demos/geomap-wet-4.0.26/geomap-en.html

**Comment by lucas-hay** [comment]
thanks @duboisp, everything appears to be working for me now.

@jvanulde Can you fix the merge conflict and we will get this merged?
We have pushed the release again as we want to ensure this gets included since the current examples are not working.

**Comment by duboisp** [comment]
@jvanulde thanks for adding the alert warning box, but the information inside and link they are pointing to are not helpful. Simply referring to the newest configuration is not enough to help web author and web developer to migrated their existing geomap instance be compatible with this update. 

I was expecting to have step by step instruction on how I can use a configuration that was working for WET 4.0.26, as per the working example, and then be fully migrate it to WET 4.0.27 by keeping the same functionality for the end user. What needs to be changed in the configuration? What information that is needed in order to obtain the same end results? What are those steps? All that detailed information should be documented in the geomap documentation page.

Web author, web developer and implementer should not need to reverse engineering the new configuration file with the previous one. That is not a good strategy at all.

In the documentation, you mention that ATOM and GeoRSS is not supported anymore. That is fine, but how that can be replaced? What is the alternative options? Suggest a few at least if there is a lot of alternative. Explain, what you did to make the working example "Static Map example" and the working example "OSM basemap example" to be compatible with this new version of Geomap.

:-)

**Comment by jvanulde** [comment]
Okay @duboisp I admit I don't know how to proceed with this. The simple fact is that the resources that were referenced in the v4.0.26 OSM basemap example no longer exist. They were 3rd party resources that were taken down by MapQuest. The fix is to simply replace the configuration with the new one. That's the only step. It's as if an image resource was removed from a directory that was referenced in an earlier version. The fix would be to point to another image in the new version. That's basically what's happening here. The examples are just that, examples. We have zero control over the data and map tiles that are used in them. I don't know how to describe the migration other than just replace the configuration. Regarding the static map example, I have created a new snippet for v4.0.27 in the documentation already. Additionally, I have pointed users to the configuration in case they want to just cut and paste it. The only thing I can think of doing is to say replace snippet A with snippet B. Not sure what else I can do.

**Comment by duboisp** [comment]
as discussed, @jvanulde would update the broken OSM basemap working example and describe how it can be replated by another basemap service that provide global map. It was noted there is a risk the global map service would be a pay services. The goal is to provide an alternate version, at least the instruction how to implement it.

**Pull request closed without merging**
